sessionctx/variable/sysvar.go

```diff
@@ -718,6 +718,14 @@ var defaultSysVars = []*SysVar{
 			return nil
 		},
 	},
+	{Scope: ScopeGlobal, Name: TiDBMemOOMAction, Value: DefTiDBMemOOMAction, PossibleValues: []string{"CANCEL", "LOG"}, Type: TypeEnum,
```

Comment:
For now the `Validate()` function of this sysvar is empty, so the following statement will be accepted:

```
set global tidb_mem_oom_action=true;
```
Although it doesn't change anything.

Please add the validate function to refine this behavior, you could follow the existing implementation like [this](https://github.com/pingcap/tidb/blob/9db3590badd97e5229c19f9bcc9331b05ce3550f/sessionctx/variable/sysvar.go#L1327-L1331)

Reply:
As Morgan mentioned the behavior is intentional and compatible with mysql.

Reply:
Here is an example from MySQL 8.0 showing ordinal value setting:

```sql
mysql [localhost:8028] {msandbox} ((none)) > select @@transaction_isolation;
+-------------------------+
| @@transaction_isolation |
+-------------------------+
| READ-COMMITTED          |
+-------------------------+
1 row in set (0.00 sec)

mysql [localhost:8028] {msandbox} ((none)) > set transaction_isolation = 2;
Query OK, 0 rows affected (0.00 sec)

mysql [localhost:8028] {msandbox} ((none)) > select @@transaction_isolation;
+-------------------------+
| @@transaction_isolation |
+-------------------------+
| REPEATABLE-READ         |
+-------------------------+
1 row in set (0.00 sec)
```

The code for `TiDBPartitionPruneMode` should be changed to use `TypeEnum` as well. Currently it looks like it does not handle values in a case insensitive way - but in MySQL it is expected that they should be.

Reply:
As Morgan mentioned existing  behavior is intentional and compatible with mysql. 

@bb7133 , As per your suggestion tweaked existing framework to perform validation on ordinal values where ever applicable( to maintain consistency across the board).
Please find the attached patch for the same.
[allow-ordinal-value.txt](https://github.com/pingcap/tidb/files/8710699/allow-ordinal-value.txt)
.

Reply:
@ramanich1 Thank you for your effort, I didn't realize that this is intentioal(because of `TypeEnum` as Morgan mentioend), so that's fine and we do not need to update it.